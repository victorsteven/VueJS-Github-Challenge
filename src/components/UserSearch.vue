<template>
  <div>
    <div class="wrapper_input">
      <form @submit.prevent="$emit('submitted', username)">
        <input
          type="text"
          class="input_search"
          placeholder="Search for Github User..."
          v-model="username"
          />
        <button type="submit" class="button_search">
          Search
        </button>
      </form>
    </div>
  </div>
</template>
