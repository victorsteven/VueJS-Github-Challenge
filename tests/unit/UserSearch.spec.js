import { shallowMount } from '@vue/test-utils';
import UserSearch from '../../src/components/UserSearch.vue';

describe('UserSearch', () => {
  const util = () => {
    const wrapper = shallowMount(UserSearch);

    return {
      wrapper,
      input: () => wrapper.find('input'),
      button: () => wrapper.find('button'),
    };
  };

  it('renders the UserSearch component', () => {
    const { wrapper } = util();

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('Assert that we have an input field and a search button', () => {
    const { input, button } = util();

    expect(input().exists()).toBe(true);
    expect(button().exists()).toBe(true);
  });

  it('calls "submitted" event when submitting form', () => {
    // arrange
    const expectedUser = 'victorsteven';
    const { wrapper, button, input } = util();
    input().element.value = expectedUser;

    input().trigger('input');
    button().trigger('click');
    button().trigger('submit');

    expect(wrapper.emitted().submitted[0]).toEqual([expectedUser]);
  });
});
