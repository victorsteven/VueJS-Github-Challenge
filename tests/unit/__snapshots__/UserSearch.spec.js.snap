// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserSearch renders the UserSearch component 1`] = `
<div>
  <div class="wrapper_input">
    <form><input type="text" placeholder="Search for Github User..." class="input_search"> <button type="submit" class="button_search">
        Search
      </button></form>
  </div>
</div>
`;

exports[`UserSearch renders the component 1`] = `
<div>
  <div class="wrapper_input">
    <form><input type="text" placeholder="Search for Github User..." class="input_search"> <button type="submit" class="button_search">
        Search
      </button></form>
  </div>
</div>
`;
